File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = {}
        buckets = [[] for _ in range(len(nums) + 1)]
        res = []

        for n in nums:
            if n in counts:
                counts[n] += 1
            else:
                counts[n] = 1
        
        for num,count in counts.items():
            buckets[count].append(num)
        
        for i in range(len(buckets) - 1, -1, -1):
            for val in buckets[i]:
                res.append(val)
                if len(res) == k:
                    return res
        
        return res

File: Data Structures & Algorithms/house-robber/submission-3.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if(nums.size() < 2){
            return nums[0];
        }
        vector<int> dp = nums;
        dp[0] = nums[0];
        dp[1] = max(nums[0], nums[1]);
        for(int i = 2; i < nums.size(); i++){
            dp[i] = max(dp[i-1], dp[i-2] + dp[i]);
        }

        return max(dp[dp.size()-1], dp[dp.size()-2]);
    }
};
